### PR TITLE
feat(Memrise): add presence

### PR DESCRIPTION
**Pull request by Extiriority**
## Description 
A rich presence addon for Memrise, A language learning platform where you can learn many vocabularies, grammar pronunciations and many more.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
</a>
</a>
</a>
</a>

</details>

### CODE REVIEW

File: websites/M/Memrise/presence.ts [hunk 1]
```diff
@@ -0,0 +1,254 @@
+const presence = new Presence({ clientId: "1095377958241304586" }),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+enum Assets {
+	Logo = "https://i.imgur.com/emLcjZT.png",
+	Searching = "https://i.imgur.com/OIgfjTG.png",
```
Comment (theusaf) [inline websites/M/Memrise/presence.ts:6]: If possible, make this one 512x512px too.

File: websites/M/Memrise/presence.ts [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+	details?: string;
+	smallImageKey?: string;
+	largeImageKey?: string;
+	state?: string;
+	buttons?: [ButtonData, ButtonData?];
+} {
+	switch (page) {
+		case "dashboard":
+			return {
+				details: `Viewing their ${page}...`,
```
Comment (theusaf) [inline websites/M/Memrise/presence.ts:47]: Prefer strings without `...`. (Apply to other strings as well)
```suggestion
				details: `Viewing their ${page}`,
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/M/Memrise/presence.ts (hunk 1 above)
Comment (Extiriority) [reply websites/M/Memrise/presence.ts:6]: it's already 512x512 or higher

Comment (Bas950) [inline websites/M/Memrise/presence.ts:6]: Please use 512x512, I will update the docs page in a sec, I think the README in this repository already has that it must be exactly 512x512.

File: websites/M/Memrise/presence.ts [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+				state: `${document.querySelector(".rank").textContent} 
+				[Best Streak ${
+					document.querySelector("#attendance-grid-label2 .large").textContent
+				} | 
+				Current ${
+					document.querySelector("#attendance-grid-label3 .large").textContent
+				} | 
+				Total ${
+					document.querySelector("#attendance-grid-label1 .large").textContent
+				} days the last 365 days]`,
```
Comment (Bas950) [inline websites/M/Memrise/presence.ts:218]: You sure this sentence is correct?

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/M/Memrise/presence.ts (hunk 1 above)
Comment (Extiriority) [reply websites/M/Memrise/presence.ts:6]: Just like Darkvillager said, Thats the resourses standards height & width. Unless the default icons had beeen changed too.

File: websites/M/Memrise/presence.ts (hunk 3 above)
Comment (Extiriority) [reply websites/M/Memrise/presence.ts:218]: Yes, should be the total numbers of days you have learned the past 365 days. So in short: "Total 36 days the last 365 days". Unless there is a better sentence.

File: websites/M/Memrise/presence.ts (hunk 1 above)
Comment (Bas950) [reply websites/M/Memrise/presence.ts:6]: Oh alright, yea didn't know they weren't, I have updated them in https://github.com/PreMiD/Presences/commit/28cd6f6f8f5e7318cd109de4ecc5552c93da4683

File: websites/M/Memrise/presence.ts (hunk 3 above)
Comment (Bas950) [reply websites/M/Memrise/presence.ts:218]: Maybe add `in` so it is `... days in the last 365 days`

Comment (Extiriority) [reply websites/M/Memrise/presence.ts:218]: Good one, I'll change it soon

**Comment by Bas950** [comment]
> @Bas950 this pr #7299 only had deepscan and not lint check so that presence screaming at me. How shall we proceed?

Fixed

File: websites/M/Memrise/presence.ts [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+						url: `https://app.memrise.com/course/${pageDetail}/${title}`,
+					},
+				],
+			};
+		case "aprender": {
+			const translate = document.querySelector(
+					"#__next > div > div > div > div > div > div > div > div > div > div > div > h2"
+				),
+				points = document.querySelector(
+					"#__next > div > div > div > div > div > div > div > div > div > span > div"
```
Comment (DarkVillager) [inline websites/M/Memrise/presence.ts:69]: Try to shorten selectors where possible. (So its more stable when changed.)
```suggestion
					'[data-testid="growthLevelIndicator"]'
```

File: websites/M/Memrise/presence.ts [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+					{
+						label: "Go to Course",
+						url: `https://app.memrise.com/course/${pageDetail}/${title}`,
+					},
+				],
+			};
+		case "aprender": {
+			const translate = document.querySelector(
+					"#__next > div > div > div > div > div > div > div > div > div > div > div > h2"
+				),
```
Comment (DarkVillager) [inline websites/M/Memrise/presence.ts:67]: Try to shorten selectors where possible. (So its more stable when changed.)
```suggestion
		case "aprender": {
			const translate = document.querySelector('[data-testid="testLearnableCard"]').querySelectorAll("h2")[1],
```
<br></br>
_If you want to go extreme to make sure you get the right element every time then you could use this (Not recommended)_
```suggestion
		case "aprender": {
const toTranslate = Array.prototype.slice.call(
				document
					.querySelector('[data-testid="testLearnableCard"]')
					?.querySelectorAll("h2") ?? []
			),
			translate =
				toTranslate.length > 1
					? toTranslate.filter((x: Element) => {
							if (
								x &&
								!x
									.getAttribute("data-testid")
									?.toLowerCase()
									?.includes("instruction")
							)
								return x.textContent;
							else return null;
					  })
					: null;
```

File: websites/M/Memrise/presence.ts [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+				points = document.querySelector(
+					"#__next > div > div > div > div > div > div > div > div > div > span > div"
+				),
+				completed = document.querySelector(
+					"#__next > div > div > div > div > div > div > div > h2"
+				);
```
Comment (DarkVillager) [inline websites/M/Memrise/presence.ts:73]: Try to shorten selectors where possible. (So its more stable when changed.)
```suggestion
				completed = document.querySelector('[data-testid="scenario-header"]')?.closest("div");
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/Memrise/presence.ts (hunk 4 above)
Comment (Extiriority) [reply websites/M/Memrise/presence.ts:69]: Will try, their website specifically on that is a bit tricky

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by DarkVillager — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**